### PR TITLE
Add some more documentation and samples to RFW

### PR DESCRIPTION
**Pull request by Hixie**
## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [ ] I signed the [CLA].
- [ ] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [ ] I [linked to at least one issue that this PR fixes] in the description above.
- [ ] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [ ] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[linked to at least one issue that this PR fixes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Comment by Hixie** [comment]
(this is not yet ready to land as-is; publishing this to get feedback)

File: packages/rfw/lib/src/flutter/core_widgets.dart [hunk 1]
```diff
@@ -64,56 +61,82 @@ import 'runtime.dart';
...
 /// that take structured types are represented using maps, with each named
-/// parameter of that type's default constructor represented by a key, with the
-/// following notable caveats and exceptions:
+/// parameter of that type's default constructor represented by a key.
+///
+/// For example, a [TextStyle] argument of a [Text] widget would look like this:
+///
+/// ```rfwtxt
+/// Text(
+///   text: 'hello',
```
Comment (ScottS2017) [inline packages/rfw/lib/src/flutter/core_widgets.dart:70]: Text takes a ListString>:
`'Text': (BuildContext context, DataSource source) {
    String? text = source.v<String>(['text']);
    if (text == null) {
      final StringBuffer builder = StringBuffer();
      final int count = source.length(['text']);
      for (int index = 0; index < count; index += 1) {
        builder.write(source.v<String>(['text', index]) ?? '');
      }
      text = builder.toString();
    }`

**Comment by stuartmorgan** [comment]
@Hixie should this still be open? It's been unchanged in Draft for several months.

**Comment by Hixie** [comment]
We can close it, I pushed it to show Scott.

**Pull request closed without merging**